src/de.hpi.swa.graal.squeak/src/de/hpi/swa/graal/squeak/nodes/ResumeContextNode.java

```diff
@@ -41,6 +46,20 @@ protected final Object doResumeInMiddle(final ContextObject context) {
         return executeContextNode.executeResumeInMiddle(context.getTruffleFrame(), initialPC);
     }
 
+    @Fallback
```

Comment:
I'm going to replace `@Fallback` with:
`@Specialization(guards = "context.getInstructionPointerForBytecodeLoop() < 0")`.

The generated fallback guard would currently check  the PC value twice (`== 0` and then `> 0`).